bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java

```diff
@@ -490,10 +490,10 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
             logger.debug("Config description validation exception occurred for thingUID {} - Messages: {}", thingUID,
                     ex.getValidationMessages());
             return Response.status(Status.BAD_REQUEST).entity(ex.getValidationMessages(locale)).build();
-        } catch (IllegalArgumentException ex) {
-            logger.info("Received HTTP PUT request for update config at '{}' for the unknown thing '{}'.",
-                    uriInfo.getPath(), thingUID);
-            return getThingNotFoundResponse(thingUID);
+        } catch (Exception ex) {
+            logger.info("Exception during HTTP PUT request for update config at '{}' for thing '{}': {}",
```

Comment:
As suggested, better use error logging to clearly indicate that we stumbled across an implementation bug.